cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -278,6 +281,18 @@ def validate_networking_requirements(
                 FailureLevel.WARNING,
             )
 
+    def validate_allocation_strategy(self, compute_resource_name: str, capacity_type: Enum, allocation_strategy: Enum):
+        """On-demand Capacity type only supports "lowest-price" allocation strategy."""
+        if (
+            capacity_type == cluster_config.CapacityType.ONDEMAND
+            and allocation_strategy == cluster_config.AllocationStrategy.CAPACITY_OPTIMIZED
```

Comment:
change to `allocation_strategy` different to `LOWEST_PRICE` so that if a new allocation strategy is added and not supported by ONDEMAND, the check will be still valid